### ISSUE TITLE
Duplicating a block input after rename doesn't Keep Rename

### ISSUE THREAD

**Issue by cycomachead** [jmoenig/Snap--Build-Your-Own-Blocks#853]
to reproduce:
1. Make a block with an input
2. drag out a copy of that input
3. right click and rename
4. right click to duplicate

Notice that the duplicated variable has the original name.

**Comment by cycomachead** [comment]
Some additional notes

1. Try "relabeling" a block with a duplicated input to see how the variable names don't match.
2. Clicking "Apply" _seems_ to fix the issue, so that duplicating now keeps the correct name -- but it doesn't fix the issue when relabeling.

I still have this weird hunch something here might be related to #712 but I don't know for sure...

**Comment by jmoenig** [comment]
yep, it's a bug. I'm sure it is relate to our recent duplication optimization effort. Thanks for reporting it!

**Comment by cycomachead** [comment]
> . I'm sure it is related to our recent duplication optimization effort.

I checked in a pre-optimization branch (s.b.e/snapsource/stable, and the OOP one) and it's present in those cases as well, same steps to reproduce, so I think there's something specific in the area of duplicating block inputs.

**Comment by jmoenig** [comment]
Thanks for checking! This should be easy to fix
